app/models/user.rb

```diff
@@ -0,0 +1,15 @@
+class User < ApplicationRecord
```

Comment:
The `User` model defines `COMPARE_LIMIT_FOR_NON_AUTH` and `COMPARE_LIMIT_FOR_AUTH` as constants but then uses them in a method to return an `OpenStruct`. Consider using these constants directly where needed instead of wrapping them in a method.